program/rust/src/utils.rs

```diff
@@ -123,3 +126,28 @@ pub fn read_pc_str_t(source: &[u8]) -> Result<&[u8], ProgramError> {
         }
     }
 }
+
+fn valid_writable_account(program_id: &Pubkey, account: &AccountInfo, minimum_size: usize) -> bool {
+    account.is_writable
+        && account.owner == program_id
+        && account.data_len() >= minimum_size
+        && Rent::default().is_exempt(account.lamports(), account.data_len())
+}
+
+pub fn check_valid_writable_account(
+    program_id: &Pubkey,
+    account: &AccountInfo,
+    minimum_size: usize,
+) -> Result<(), ProgramError> {
+    pyth_assert(
+        valid_writable_account(program_id, account, minimum_size),
+        OracleError::InvalidWritableAccount.into(),
+    )
+}
+
+pub fn is_component_update(cmd_args: &cmd_upd_price_t) -> Result<bool, ProgramError> {
```

Comment:
why "component"? 

also, the argument to this is confusing because if you have a `cmd_upd_price_t`, you would expect its cmd_ field to be the corresponding int.

Reply:
Component as opposed to aggregate.

Reply:
This one is tricky : the dispatcher routes `upd_price`, `upd_price_no_fail_on_error` and `agg_price` to this function . The first two update the individual publisher's price, unlike `agg_price` which only tries to update the aggregate.

Reply:
ah gotcha. this is fine